### ISSUE TITLE
[Feature Request] Parallel creation of (many) TCP/IP Tunnels

### ISSUE THREAD

**Issue by OOTS** [ParallelSSH/parallel-ssh#187]
Hi,

I'd like to use parallel-ssh to establish SSH tunnels (port forwarding) to (many) hosts. Right now, you can establish each tunnel manually using the `Tunnel` class in `pssh.clients.native.tunnel`. However, it would be cool to have a feature like the following:

```python

hosts=[...]
client = ParallelSSHClient(hosts, other_args)
ports = client.open_tunnel(remote_host, remote_port, bind_ip)
# now, all traffic (from bind_ip) to localhost:ports[i] is sent to hosts[i], which forwards it to remote_host:remote_port
```

What I'm trying to do is the following:
I have a couple of hosts, each running an SSHD and the same set of services. The services only listen for connections from localhost (by intention). The current way of connecting to them is to establish a port forwarding through the SSHD to the respective service ports with `ssh -N -L localhost:local_port:localhost:service_port user@host`. I'd like to be able to establish one tunnel for each host with a simple API.

Best regards,
OOTS

**Comment by pkittenis** [comment]
Thank you for the interest. That would be nice to have, though no ETA at the moment. Related for performance enhancement of tunnelling - #140

**Comment by pkittenis** [comment]
So to understand correctly, the intention with `ports = client.open_tunnel(remote_host, remote_port, bind_ip)` is to have each host in `client.hosts` connect to `remote_host:remote_port` and tunnel that traffic to the client's localhost:<a port>.

Making a parallel API for that should not be too much work, after #239.

**Comment by OOTS** [comment]
Hi,

my intention was to have the call `ports = client.open_tunnel(remote_host, remote_port, bind_ip)` achieve the following:
The client (who is running the Python code) opens a couple of listening ports (returned as `ports`). Any incoming connection to `ports[i]` is tunneled through the SSH connection to `hosts[i]`, and from there forwarded to `(remote_host, remote_port)` (as seen from `hosts[i]`).
This was supposed to recreate the behaviour of `ssh -L $bind_ip:$ports[i],$remote_host,$remote_port $hosts[i]`.

However, by now I (personally) don't need this feature anymore, I've got a working alternative. But maybe someone else will find this feature useful.

**Comment by pkittenis** [comment]
Yes, that's what I meant. There is already functionality to do that and is used for client proxying, though no nice parallel API for ad-hoc tunnels.

Providing this functionality is not part of this [project's goals](https://github.com/ParallelSSH/parallel-ssh#design-and-goals), and capability exists for users to do it pretty easily already - see `pssh.clients.native.tunnel`.

So closing. If someone else needs it and is up for writing a parallel function I'd be happy to review a PR.
